app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -55,85 +55,47 @@ def pulp_data(pulp_id)
-          update_repository_associations(results) if index_repository_association
         end
       end
 
-      def import_for_repository(repository, force = false)
-        ids = content_unit_class.ids_for_repository(repository.pulp_id)
-        # Rpms cannot change in Pulp so we do not index them if they are already present
-        # in our database. Errata and Package Groups can change in Pulp, so we index
-        # all of them in the repository on each sync.
-        if immutable_unit_types.include?(self) && !force
-          ids_to_import = ids - repository.rpms.map(&:pulp_id)
-        else
-          ids_to_import = ids
-        end
-        self.import_all(ids_to_import, :index_repository_association => false) if repository.content_view.default? || force
-        self.sync_repository_associations(repository, :pulp_ids => ids) if self.manage_repository_association
-      end
-
-      def unit_id_field
-        "#{self.name.demodulize.underscore}_id"
-      end
-
-      def copy_repository_associations(source_repo, dest_repo)
-        if manage_repository_association
-          delete_query = "delete from #{repository_association_class.table_name} where repository_id = #{dest_repo.id} and
-                         #{unit_id_field} not in (select #{unit_id_field} from #{repository_association_class.table_name} where repository_id = #{source_repo.id})"
-
-          insert_query = "insert into #{repository_association_class.table_name} (repository_id, #{unit_id_field})
-                          select #{dest_repo.id} as repository_id, #{unit_id_field} from #{repository_association_class.table_name}
-                          where repository_id = #{source_repo.id} and #{unit_id_field} not in (select #{unit_id_field}
-                          from #{repository_association_class.table_name} where repository_id = #{dest_repo.id})"
-        else
-          columns = column_names - ["id", "pulp_id", "created_at", "updated_at", "repository_id"]
-
-          delete_query = "delete from #{self.table_name} where repository_id = #{dest_repo.id} and
-                          pulp_id not in (select pulp_id from #{self.table_name} where repository_id = #{source_repo.id})"
-          insert_query = "insert into #{self.table_name} (repository_id, pulp_id, #{columns.join(',')})
-                    select #{dest_repo.id} as repository_id, pulp_id, #{columns.join(',')} from #{self.table_name}
-                    where repository_id = #{source_repo.id} and pulp_id not in (select pulp_id
-                    from #{self.table_name} where repository_id = #{dest_repo.id})"
-
+      def import_for_repository(repository)
+        pulp_ids = []
+        service_class = SmartProxy.pulp_master!.content_service(content_type)
+        service_class.pulp_units_batch_for_repo(repository).each do |units|
+          units.each do |unit|
+            unit = unit.with_indifferent_access
+            unit['_id'] = unit['_href'] ? unit['_href'] : unit['_id']
```

Comment:
This logic won't work because pulp2 units have an '_href' as well, but we don't use it.  What about doing something like 'service_class.unit_identifier' and have that return '_id' or '_href' ?